src/external_calls.rs

```diff
@@ -57,6 +57,7 @@ pub fn run<
 ) -> (
     SchedulerCircuitInstanceWitness<MainField, CircuitGoldilocksPoseidon2Sponge, GoldilocksExt2>,
     BlockAuxilaryOutputWitness<MainField>,
+    S,
```

Comment:
This function is used here:

https://github.com/matter-labs/zksync-era/blob/095bc243b413851a2fe77b8cab512ce1e851054d/prover/witness_generator/src/basic_circuits.rs#L585

Maybe changed interface will cause problems there

Reply:
Actually imo it is better to customize `run_vms` instead of this function and use it in `compiler_tests_runner.rs`

Reply:
Yeah, makes sense, I was thinking that this function is used only for the tests

Reply:
Will fix it, thanks